frontend/src/services/objectService.ts

```diff
@@ -227,7 +228,54 @@ export default {
   searchObjects(params: SearchObjectsOptions = {}, headers: any = {},) {
     // remove objectId array if its first element is undefined
     if (params.objectId && params.objectId[0] === undefined) delete params.objectId;
-    return comsAxios().get(PATH, { params: params, headers: headers });
+
+    if (params.objectId) {
+      /**
+       * split calls to COMS if query params (eg objectId's)
+       * will cause url length to excede 2000 characters
+       * see: https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers
+       *
+       * TODO: consider creating a utils function
+       * eg: `divideParam(params, attr)`
+       *     ... return Promise.all(divideParam(params, objectId).map(zparam => comsAxios().get(PATH, {params: zparam, headers: headers});
+       */
+      let urlLimit = 2000;
+      // minus base url
+      const baseUrl = new URL(`${new ConfigService().getConfig().coms.apiPath}${PATH}`).toString().length;
+      urlLimit = urlLimit - baseUrl;
+      // minus deleteMarker=false
+      urlLimit = urlLimit - 19;
+      // minus latest=false
+      urlLimit = urlLimit - 13;
+      // minus a single bucketId (bucketId[]=<uuid>)
+      if (params.bucketId) { urlLimit = urlLimit - 48; }
```

Comment:
- May be worth rewriting these using subtraction assignment instead as it'll make it more concise.
- The const assignment on Line 244 should be the baseUrl string, not the number value thereof.
- The conditional body on Line 251 doesn't need curly brace scoping as it's just a one-liner.

```ts
const baseUrl = new URL(`${new ConfigService().getConfig().coms.apiPath}${PATH}`).toString();
urlLimit -= baseUrl.length; // subtract baseUrl length
urlLimit -= 19; // subtract `deleteMarker=false`
urlLimit -= 13; // subtract `latest=false`
if (params.bucketId) urlLimit -= 48; // subtract a single bucketId `bucketId[]=<uuid>`
```

Reply:
Will parameters like `deleteMarker=false` and `latest=false` be guaranteed to exist? Or should some logic be used to check for them before subtracting the limit as future proofing?